index.js

```diff
@@ -527,14 +527,18 @@ const plugin = fp(async function (app, opts) {
     }
 
     // Trigger preExecution hook
+    let modifiedSchema
     let modifiedDocument
     if (context.preExecution !== null) {
-      ({ modifiedDocument } = await preExecutionHandler({ schema: fastifyGraphQl.schema, document, context }))
+      ({ modifiedSchema, modifiedDocument } = await preExecutionHandler({ schema: fastifyGraphQl.schema, document, context }))
     }
 
     // minJit is 0 by default
     if (shouldCompileJit) {
-      cached.jit = compileQuery(fastifyGraphQl.schema, modifiedDocument || document, operationName)
+      cached.jit = compileQuery(
+        modifiedSchema || fastifyGraphQl.schema,
+        modifiedDocument || document, operationName
+      )
```

Comment:
I don't think we should cache or compile queries where the schema has been dynamically modified.

There is definitely a race here where a schema is modified and then cached while a subsequent query would change it in a different way (maybe on user permission).